Introduce 'rotateRadians' helper function and use across WebKit

Ahmad-S792 opened the pull request:
#### 3f64447f2aa9d256ae17bd271e186aac0257f2f9
<pre>
Introduce &apos;rotateRadians&apos; helper function and use across WebKit

<a href="https://bugs.webkit.org/show_bug.cgi?id=262718">https://bugs.webkit.org/show_bug.cgi?id=262718</a>

Reviewed by Chris Dumez.

Merge: <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=167603 &amp;
<a href="https://src.chromium.org/viewvc/blink?revision=168929&amp">https://src.chromium.org/viewvc/blink?revision=168929&amp</a>;view=revision

Currently, rotate function takes an argument in only degrees.
In many cases, rotate function is often required radian value instead of degree value.
If we add new rotateRadians function, we will be able to reduce unnecessary operations.

 Before : rad -&gt; rad2deg -&gt; deg2rad -&gt; calc sinNcos
 After  : rad                       -&gt; calc sinNcos

As a result, we can reduce two multiplication and two division.

Additionally, we can use it across WebCore source instead of &apos;rotate(rad2deg)&apos;.

* Source/WebCore/platform/graphics/transforms/AffineTransform.cpp:
(AffineTransform::rotate):
(AffineTransform::rotateRadians):
(AffineTransform::rotateFromVector):
(AffineTransform::decompose):
* Source/WebCore/platform/graphics/transforms/AffineTransform.h: new &apos;rotateRadians&apos; function
* Source/WebCore/platform/graphics/cg/PathCG.cpp:
(PathCG::addEllipse):
* Source/WebCore/html/canvas/CanvasPath.cpp:
(CanvasPath::addEllipse):
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp:
(TransformationMatrix::rotateFromVector):
(TransformationMatrix::rotate):
(TransformationMatrix::rotateRadians):
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.h: new &apos;rotateRadians&apos; function

Canonical link: <a href="https://commits.webkit.org/268964@main">https://commits.webkit.org/268964@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/683dfbcb41aac70e2d1fd0e63222b3d307323edc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21154 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21508 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22212 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23025 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19661 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21394 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24774 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21705 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21378 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21083 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18341 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23878 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18240 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19172 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25507 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19321 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19373 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23368 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19907 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16927 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19189 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5073 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23475 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19770 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
Still LGTM if the bots are happy.

Ahmad-S792 committed:
Introduce 'rotateRadians' helper function and use across WebKit

https://bugs.webkit.org/show_bug.cgi?id=262718

Reviewed by Chris Dumez.

Merge: https://src.chromium.org/viewvc/blink?view=revision&revision=167603 &
https://src.chromium.org/viewvc/blink?revision=168929&view=revision

Currently, rotate function takes an argument in only degrees.
In many cases, rotate function is often required radian value instead of degree value.
If we add new rotateRadians function, we will be able to reduce unnecessary operations.

 Before : rad -> rad2deg -> deg2rad -> calc sinNcos
 After  : rad                       -> calc sinNcos

As a result, we can reduce two multiplication and two division.

Additionally, we can use it across WebCore source instead of 'rotate(rad2deg)'.

* Source/WebCore/platform/graphics/transforms/AffineTransform.cpp:
(AffineTransform::rotate):
(AffineTransform::rotateRadians):
(AffineTransform::rotateFromVector):
(AffineTransform::decompose):
* Source/WebCore/platform/graphics/transforms/AffineTransform.h: new 'rotateRadians' function
* Source/WebCore/platform/graphics/cg/PathCG.cpp:
(PathCG::addEllipse):
* Source/WebCore/html/canvas/CanvasPath.cpp:
(CanvasPath::addEllipse):
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp:
(TransformationMatrix::rotateFromVector):
(TransformationMatrix::rotate):
(TransformationMatrix::rotateRadians):
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.h: new 'rotateRadians' function

Canonical link: https://commits.webkit.org/268964@main
